### DISCUSSION TITLE
Automatic Node Pickup.

### DISCUSSION THREAD

**Discussion by nnitch** [rqlite/rqlite#1739] [Q&A] [answered]
Hey, 

I'm looking at using rqlite in one of my projects, and just want to ask a couple questions.

### Scenario One - Node catchup after localised outage.
I have a cluster of 5 nodes, all with the exact same database, containing the same data. All nodes in a rqlite cluster. During operation, one of the 5 nodes fails, and is completely disconnected from the rqlite cluster. During this nodes downtime, many writes are made to the cluster, written to all nodes except the one that is currently offline.

Eventually, the node that that was previously offline, comes back online. 

**Does the writes that were previously missed get pushed to the database so that the database actually reflects the data that the other nodes do?**

### Scenario Two - Adding nodes to cluster with ZERO DOWNTIME.
I have a cluster of 5 nodes. Everything is perfect, but I'd like to expand to increase resilience by adding some more nodes. Keep in mind that the database cluster is receiving roughly one write a second. Will the writes that occur to the cluster be written to the database once replication has been completed?

In my eyes, the only way to prevent data from being mismatched between "old" and "new" nodes is to have some sort of downtime to prevent data being written to the old nodes before the new nodes are integrated into the cluster with the same replicated database.

**Comment by otoolep** [comment, marked as answer]
Have you reviewed the docs on rqlite.io?

>I have a cluster of 5 nodes, all with the exact same database, containing the same data.

This is actually how rqlite works, it cannot operate in any other way. That you feel you need to explicitly mention it makes me wonder if you think rqlite can operate in a mode where this is not true. I just want to check.

>Does the writes that were previously missed get pushed to the database so that the database actually reflects the data that the other nodes do?

Yes, but how are you expecting to detect if this is not the case? Have you reviewed https://rqlite.io/docs/api/read-consistency/ ?

>Will the writes that occur to the cluster be written to the database once replication has been completed?

Yes, but again I suggest you review the Read Consistency doc above. In rqlite all reads are served, by default, by the leader. I get the impression from your questions (which are reasonable questions, don't get me wrong) that you don't have a clear mental model of how rqlite serves reads.
